init_scripts/00_regen_sharelatex_secrets.sh

```diff
@@ -1,7 +1,7 @@
 #!/bin/sh
 # Create random secret keys (twice, once for http auth pass, once for cookie secret).
 CRYPTO_RANDOM=$(dd if=/dev/urandom bs=1 count=32 2>/dev/null | base64 -w 0 | rev | cut -b 2- | rev | tr -d '\n+/')
-sed -i "0,/CRYPTO_RANDOM/s/CRYPTO_RANDOM/$CRYPTO_RANDOM/" /etc/sharelatex/settings.coffee
+echo "export WEB_API_PASSWORD=$CRYPTO_RANDOM" > /etc/profile.d/crypto.sh
 
 CRYPTO_RANDOM=$(dd if=/dev/urandom bs=1 count=32 2>/dev/null | base64 -w 0 | rev | cut -b 2- | rev | tr -d '\n+/')
-sed -i "0,/CRYPTO_RANDOM/s/CRYPTO_RANDOM/$CRYPTO_RANDOM/" /etc/sharelatex/settings.coffee
+echo "export CRYPTO_RANDOM=$CRYPTO_RANDOM" >> /etc/profile.d/crypto.sh
```

Comment:
This will regenerate the session secret on every boot up and hence invalidate all sessions from the previous run.

Reply:
Yeah, we're not changing that behaviour (although any suggestion is always welcome)

Reply:
I don't think we should make a radical change to the inter-reboot behaviour at this time, but it's good to know it can be done :)

Reply:
I assume that the `/profile.d/crypto.sh` file is loaded in the environment for each service. Have we verified that this is the case, and that the values change between runs?

Reply:
I also note that the second invocation uses `>>` to append the new value to the `/etc/profile.d/crypto.sh` file, while the first invocation replaces the file outright. I _think_ this will amount to the desired behaviour, but it's woth noting the append here.

Following on from @das7pad 's comment, did the previous setup preserve sessions between reboots? I can't quite remember. Either way we should preserve that behaviour, or make a note of it in the release notes if it absolutely must change.

Reply:
Yes it did preserve the secret: a third sed run on the file (e.g. after a reboot) will not see any `CRYPTO_RANDOM` references anymore.

Reply:
Ok, in that case I think we should maybe try to preserve that behaviour. How about something like this (in pseudocode)? :

```
if (not exists /profile.d/crytpo.sh) {
  echo "export CRYPTO_RANDOM=${TOKEN_ONE} \n export WEB_API_PASSWORD=${TOKEN_TWO}" > /profile.d/crypto.sh
}
```

Then it should be persisted between reboots

Reply:
Here's some actual shell script which would do the thing:

```bash
if [ ! -f ./example.sh ]; then
  cat <<EOF > ./example.sh
    export XX=1 YY=2
  EOF
fi
```

Reply:
Sorted, it's checking for the files, won't be regenerated on restart

Reply:
For future reference: when restarting docker, container's filesystem is persisted and restored once docker is back, regardless of the `restart: always` configuration. This was my misunderstanding, I assumed secret recreation happened more often in current installation.